### PR TITLE
Merge 1.7 into master

### PR DESCRIPTION
**Pull request by SamirBoulil**
**Description (for Contributor and Core Developer)**

- Reworked some fixes
- merge the 1.7 integration tests refacto back into master

_**Please review all the commits after the merge 1.7 into master commit**_

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | -
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | Ok
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/controller/product.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ define(
                             return;
                         }
 
-                        PageTitle.set({'product.sku': _.escape(product.meta.label[UserContext.get('catalogLocale')]) });
+                        PageTitle.set({'product.sku': product.meta.label[UserContext.get('catalogLocale')] })
```
Comment (SamirBoulil) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/controller/product.js]: We don't need to escape characters so they appear in the page title.

File: src/Pim/Bundle/ApiBundle/EventSubscriber/CheckHeadersRequestSubscriber.php [hunk 2]
```diff
@@ -26,12 +27,19 @@ class CheckHeadersRequestSubscriber implements EventSubscriberInterface
...
     /**
-     * @param FormatNegotiator $formatNegotiator
+     * @param FormatNegotiator      $formatNegotiator
+     * @param ContentTypeNegotiator $contentTypeNegotiator
      */
-    public function __construct(FormatNegotiator $formatNegotiator)
-    {
+    public function __construct(
+        FormatNegotiator $formatNegotiator,
+        ContentTypeNegotiator $contentTypeNegotiator = null
```
Comment (ahocquard) [inline src/Pim/Bundle/ApiBundle/EventSubscriber/CheckHeadersRequestSubscriber.php]: Delete `= null` + update changelog

File: src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/QueryGenerator/MultipleOptionValueUpdatedQueryGenerator.php [hunk 3]
```diff
@@ -13,6 +15,26 @@
...
+     * @param NamingUtility       $namingUtility
+     * @param string              $entityClass
+     * @param string              $field
+     * @param NormalizerInterface $attributeOptionNormalizer
+     */
+    public function __construct(
+        NamingUtility $namingUtility,
+        $entityClass,
+        $field,
+        NormalizerInterface $attributeOptionNormalizer = null
```
Comment (ahocquard) [inline src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/QueryGenerator/MultipleOptionValueUpdatedQueryGenerator.php]: same

File: src/Pim/Bundle/ApiBundle/EventSubscriber/CheckHeadersRequestSubscriber.php [hunk 4]
```diff
@@ -60,40 +71,44 @@ public function onKernelRequest(GetResponseEvent $event)
...
                 }
 
                 return;
             }
 
             if (in_array($request->getMethod(), ['PUT', 'PATCH', 'POST'])) {
-                $contentType = $request->headers->get('content-type');
-                if (null === $contentType) {
+                $contentType = trim(strtok($request->headers->get('content-type'), ';'));
+                $allowedContentTypes = null !== $this->contentTypeNegotiator ? $this->contentTypeNegotiator->getContentTypes($request) : [];
```
Comment (ahocquard) [inline src/Pim/Bundle/ApiBundle/EventSubscriber/CheckHeadersRequestSubscriber.php]: Change it to `$allowedContentTypes =  $this->contentTypeNegotiator->getContentTypes($request);`

Comment (SamirBoulil) [inline src/Pim/Bundle/ApiBundle/EventSubscriber/CheckHeadersRequestSubscriber.php]: done

File: src/Pim/Bundle/ApiBundle/EventSubscriber/CheckHeadersRequestSubscriber.php (hunk 2 above)
Comment (SamirBoulil) [inline src/Pim/Bundle/ApiBundle/EventSubscriber/CheckHeadersRequestSubscriber.php]: done

File: src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/QueryGenerator/MultipleOptionValueUpdatedQueryGenerator.php (hunk 3 above)
Comment (SamirBoulil) [inline src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/QueryGenerator/MultipleOptionValueUpdatedQueryGenerator.php]: done

**Pull request merged**